src/hrflow_connectors/connectors/digitalrecruiters/warehouse.py

```diff
@@ -0,0 +1,356 @@
+    username: str = Field(
+        ...,
+        description="Username for authentication",
+        repr=False,
+        field_type=FieldType.Auth,
+    )
+    password: str = Field(
+        ...,
+        description="Password for authentication",
+        repr=False,
+        field_type=FieldType.Auth,
+    )
+    url_environment: HttpUrl = Field(
+        ...,
+        description="URL environment for the API",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+    jobAd: int = Field(
+        None,
+        description="Optional: Id of a job advertisement",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+    sort: str = Field(
+        None,
+        description=(
+            "Optional: Field to sort by (id, firstName, lastName, createdAt, updatedAt)"
+        ),
+        repr=False,
+        field_type=FieldType.Other,
+    )
+    limit: int = Field(
+        50,
+        description="Optional: Limit the number of results returned",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+    page: int = Field(
+        1,
+        description="Optional: Page number of results returned",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+
+
+class WriteProfilesParameters(ParametersModel):
+    digital_recruiters_token: str = Field(
+        ...,
+        description="Digital Recruiters API token.",
+        repr=False,
+        field_type=FieldType.Auth,
+    )
+    digital_recruiters_environment_url: str = Field(
+        ...,
+        description="Digital Recruiters API url environnement.",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+    job_reference: str = Field(
+        ...,
+        description="reference of the job to which the candidate is applying.",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+    message: t.Optional[str] = Field(
+        "message du candidat",
+        description="Application message.",
+        repr=False,
+        field_type=FieldType.Other,
+    )
+
+
+def get_initial_token(params: ReadProfileParameters):
+    url = f"{params.url_environment}/public/v1/users/login"
+    headers = {"Content-Type": "application/json", "X-DR-API-KEY": params.api_key}
+    payload = {"username": params.username, "password": params.password}
+
+    response = requests.post(url, headers=headers, json=payload)
+
+    if response.status_code == 200:
+        return response.json()["token"], response.json()["refresh_token"]
+    else:
+        return None, None
+
+
+def refresh_token(params: ReadProfileParameters, refresh_token: str):
+    url = f"{params.url_environment}/public/v1/users/refresh"
+
+    headers = {"Content-Type": "application/json", "X-DR-API-KEY": params.api_key}
+
+    payload = {"token": refresh_token}
+
+    response = requests.post(url, headers=headers, json=payload)
+
+    if response.status_code == 200:
+        return response.json()["token"], response.json()["refresh_token"]
+    else:
+        return None
```

Comment:
Même remarque par rapport au fait de renvoyer silencieusement `None`

En plus du genre que faudrait plutot faire `return None, None` pour pas que ca fail en mode No enough value to unpack quand tu utilise dans le code `token, refresh_token = refresh_token(...)`

Reply:
c corrigé